src/nvim/ops.c

```diff
@@ -1771,14 +1773,12 @@ int op_replace(oparg_T *oap, int c)
             memset(newp + newp_len, ' ', (size_t)bd.endspaces);
             newp_len += bd.endspaces;
             // copy the part after the changed part
-            memmove(newp + newp_len, oldp,
-                    (size_t)(oldlen - bd.textcol - bd.textlen + 1));
+            memmove(newp + newp_len, oldp, (size_t)col);
         }
       } else {
         // Replacing with \r or \n means splitting the line.
-        after_p_len = (size_t)(oldlen - bd.textcol - bd.textlen + 1);
-        after_p = (char_u *)xmalloc(after_p_len);
-        memmove(after_p, oldp, after_p_len);
```

Comment:
`after_p_len` is used below. After these changes it is always zero ...

Reply:
add
`int after_p_len = col;`?

Reply:
that seems ok. and then there's no need to change these lines.

Reply:
Now I see how I screwed up.

Reply:
fixed